ow_core/text_search_manager.php

```diff
@@ -0,0 +1,250 @@
+ * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the specific language
+ * governing rights and limitations under the License. The Original Code is Oxwall software.
+ * The Initial Developer of the Original Code is Oxwall Foundation (http://www.oxwall.org/foundation).
+ * All portions of the code written by Oxwall Foundation are Copyright (c) 2011. All Rights Reserved.
+
+ * EXHIBIT B. Attribution Information
+ * Attribution Copyright Notice: Copyright 2011 Oxwall Foundation. All rights reserved.
+ * Attribution Phrase (not exceeding 10 words): Powered by Oxwall community software
+ * Attribution URL: http://www.oxwall.org/
+ * Graphic Image as provided in the Covered Code.
+ * Display of Attribution Information is required in Larger Works which are defined in the CPAL as a work
+ * which combines Covered Code or portions thereof with code not governed by the terms of the CPAL.
+ */
+
+/**
+ * The class is responsible for text search management.
+ * 
+ * @author Alex Ermashev <alexermashev@gmail.com>
+ * @package ow.ow_core
+ * @since 1.0
+ */
+final class OW_TextSearchManager
+{
+    /**
+     * Singleton instance.
+     * @var OW_TextSearchManager
+     */
+    private static $classInstance;
+
+    /**
+     * Default storage instance     
+     * @var BASE_CLASS_InterfaceSearchStorage
+     */
+    private static $defaultStorageInstance;
+
+    /**
+     * Active storage instance     
+     * @var BASE_CLASS_InterfaceSearchStorage
+     */
+    private static $activeStorageInstance;
+
+    /**
+     * Default search storage
+     * @var string
+     */
+    private static $defaultStorage = 'BASE_CLASS_MysqlSearchStorage';
+
+    /**
+     * Returns an instance of class (singleton pattern implementation).
+     *
+     * @return OW_TextSearchManager
+     */
+    public static function getInstance()
+    {
+        if ( self::$classInstance === null )
+        {
+            self::$classInstance = new self();
+        }
+
+        return self::$classInstance;
+    }
+
+    /**
+     * Constructor.
+     */
+    private function __construct()
+    {}
+
+    /**
+     * Add entity
+     *
+     * @param string $type
+     * @param integer $id
+     * @param string  $searchText
+     * @param array $tags
+     * @param boolean $isActive
+     * @return boolean
+     */
+    public function addEntity( $type, $id, $searchText, array $tags = array(), $isActive = true )
+    {
+        $result =  self::getDefaultStorageInstance()->addEntity($type, $id, $searchText, $tags, $isActive);
+
+        if ( $result && self::getActiveStorageInstance() )
+        {
+            $result =  self::getActiveStorageInstance()->addEntity($type, $id, $searchText, $tags, $isActive);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Delete entity
+     *
+     * @param string $type
+     * @param integer $id
+     * @return boolean
+     */
+    public function deleteEntity( $type, $id )

```

Comment:
Rename param names for better method hinting: $type => $entityType, $id => $entityId.